src/forms/timeInput.test.jsx

```diff
@@ -1,49 +1,250 @@
 		name: 'time',
-		value: '22:00',
+		value: '22:12',
 		onChange: onChangePropMock,
 		required: true
 	};
 
-	beforeEach(() => {
-		onChangeSpy = jest.spyOn(TimeInput.prototype, 'onChange');
-		component = shallow(<TimeInput {...props} />);
-		inputEl = component.find('input');
-	});
+	describe('TimeInput, basic functionality', () => {
 
-	afterEach(() => {
-		component = null;
-		inputEl = null;
-		onChangeSpy.mockClear();
-	});
+		describe('getTimeParts', () => {
+			it('returns an object if no parts specified', () => {
+				const result = getTimeParts('13:00');
+				const expected = {
+					[HOURS]: '13',
+					[MINUTES]: '00',
+					[MERIDIAN]: 'PM'
+				};
+				expect(result).toEqual(expected);
+			});
+			it('returns a part if part specified', () => {
+				const result = getTimeParts('13:00', HOURS);
+				expect(result).toEqual('13');
+			});
+		});
 
-	it('renders a time html input with expected props', function() {
-		expect(component).toMatchSnapshot();
+		describe('parseValueForState', () => {
+			it('takes a value string for 24Hr time and returns an object with hours, min, meridian, value', () => {
+				const value = '13:00',
+				is24Hr = true;
+				const result = TimeInput.prototype.parseValueIntoState(value, is24Hr);
+				const expected = {
+					[HOURS]: '13',
+					[MINUTES]: '00',
+					[MERIDIAN]: false,
+					value
+				};
+				expect(result).toEqual(expected);
+			});
+			it('takes a value string, is24hr as false, and returns an object with hours, min, meridian, value', () => {
+				const value = '13:15',
+					is24Hr = false;
+				const result = TimeInput.prototype.parseValueIntoState(value, is24Hr);
+				const expected = {
+					[HOURS]: '01',
+					[MINUTES]: '15',
+					[MERIDIAN]: 'PM',
+					value
+				};
+				expect(result).toEqual(expected);
+			});
+		});
+
+		describe('constrainValue', () => {
+			it('returns the same value if no min and max', () => {
+				const value = '10';
+				const result = TimeInput.prototype.constrainValue(undefined, undefined, value);
+				expect(result).toEqual(parseInt(value));
+			});
+			it('returns the constrained value if min', () => {
+				const value = '10';
+				const min = 20;
+				const result = TimeInput.prototype.constrainValue(min, undefined, value);
+				expect(result).toEqual(min);
+			});
+			it('returns the constrained value if max', () => {
+				const value = '10';
+				const max = 5;
+				const result = TimeInput.prototype.constrainValue(undefined, max, value);
+				expect(result).toEqual(max);
+			});
+		});
 	});
 
-	it('calls internal onChange when value is changed', function() {
-		inputEl.simulate('change', { target: { value: newTime } });
-		expect(onChangeSpy).toHaveBeenCalled();
+	describe('TimeInput, with input[time] support', () => {
+		beforeEach(() => {
+			onChangeSpy = jest.spyOn(TimeInput.prototype, 'onChange');
+			onTimeInputChangeSpy = jest.spyOn(TimeInput.prototype, 'onTimeInputChange');
+			component = mount(<TimeInput {...props} />);
```

Comment:
do you need to `mount` vs `shallow`?

Reply:
yes

Reply:
Looks like it's not too tricky to enable `shallow` rendering for this:
https://github.com/meetup/meetup-web-components/blob/shallow-time-input/src/forms/timeInput.test.jsx#L119-L121
https://github.com/meetup/meetup-web-components/blob/shallow-time-input/src/forms/timeInput.test.jsx#L133

I know there's time pressure on this so it's not a blocker, but `mount` almost _always_ over-tests the component, so it should be avoided. The other uses of `mount` a little more tricky to test using `shallow`, but mainly because they make assumptions about about how `mount` renders elements - e.g. your `blur` simulate calls don't provide an argument, even though your `onBlur` handler expects an argument.

Reply:
ok. totally agree - sorry for the short response previously. I just swapped `mount` out for `shallow` and things failed - without much time to rewrite tests.  I'll put in a defect to fix tests.